client/my-sites/checkout/controller.jsx

```diff
@@ -127,15 +127,15 @@ export function conciergeSessionNudge( context, next ) {
 	);
 
 	const state = context.store.getState();
-	const selectedSite = getSelectedSite( state );
+	const selectedSiteId = getSelectedSiteId( state );
 
-	if ( ! selectedSite ) {
+	if ( ! selectedSiteId ) {
 		return null;
 	}
```

Comment:
This check should not be neccessary. The route that is handled here always has a `:site` parameter, there is no support for "All Sites" view.

And if the specified `:site` is invalid (e.g., site doesn't exist), the `siteSelection` page.js handler will detect that and won't pass control to the next handler.

The `ConciergeSessionNudge` could fetch the `selectedSiteId` from Redux in its `connect` call. The only prop that needs to come outside Redux is `receiptId`:
```jsx
<ConciergeSessionNudge receiptId={ Number( receiptId ) } />
```

Reply:
Thanks, taken care of in [9849222](https://github.com/Automattic/wp-calypso/pull/33749/commits/98492222ecd61d3624a0558dcc8394047785e873). `ConciergeSessionNudge` component also gets a `siteSlugParam ` prop, which is used to redirect a logged out user(where we have no site ID) to `/checkout/SITE_SLUG/quickstart-session` after clicking the `Get Started` button [here](https://user-images.githubusercontent.com/1269602/59347690-e1578000-8d32-11e9-9950-1d26fc7d3740.png).